File: app/templates/server/go/echo/api/todo/controller/todocontroller.go
package todocontroller

import (
	"<%= repoHostUrl %>/<%= userNameSpace %>/<%= appName %>/server/api/todo/dao"
	todo "<%= repoHostUrl %>/<%= userNameSpace %>/<%= appName %>/server/api/todo/model"
	"gopkg.in/labstack/echo.v3"
	"net/http"
)

func GetAll(c echo.Context) error {
	ts, _ := tododao.All()

	return c.JSON(http.StatusOK, ts)
}

func GetById(c echo.Context) error {
	id := c.Param("id")

	nt, _ := tododao.GetById(id)

	return c.JSON(http.StatusOK, nt)
}

func NewTodo(c echo.Context) error {
	t := new(todo.Todo)

	c.Bind(t)

	nt, _ := tododao.NewTodo(*t)

	return c.JSON(http.StatusOK, nt)
}

func RemoveTodo(c echo.Context) error {
	id := c.Param("id")

	tododao.DeleteTodo(id)

	return c.String(http.StatusOK, "")
}

File: app/templates/server/node/express/server_node_express_babel/api/todo/controller/todo-controller.js
import TodoDAO from "../dao/todo-dao";

export default class TodoController {
  static getAll(req, res) {
    TodoDAO
      .getAll()
      .then(todos => res.status(200).json(todos))
      .catch(error => res.status(400).json(error));
  }

  static getById(req, res) {
      TodoDAO
        .getById(req.params.id)
        .then(todo => res.status(200).json(todo))
        .catch(error => res.status(400).json(error));
  }

  static createTodo(req, res) {
    let _todo = req.body;

    TodoDAO
      .createTodo(_todo)
      .then(todo => res.status(201).json(todo))
      .catch(error => res.status(400).json(error));
  }

  static deleteTodo(req, res) {
    let _id = req.params.id;

    TodoDAO
      .deleteTodo(_id)
      .then(() => res.status(200).end())
      .catch(error => res.status(400).json(error));
  }
}
